### PR TITLE
Pin exact dependency versions

### PR DESCRIPTION
**Pull request by gtsonevv**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/near/near-api-js/blob/master/CONTRIBUTING.md).
- [x] Commit messages follow the [conventional commits](https://www.conventionalcommits.org/) spec
- [ ] **If this is a code change**: I have written unit tests.
- [ ] **If this changes code in a published package**: I have run `pnpm changeset` to create a `changeset` JSON document appropriate for this change.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Related issues/PRs

https://github.com/near/near-api-js/issues/1200

### CODE REVIEW

File: packages/biometric-ed25519/package.json [hunk 1]
```diff
@@ -13,19 +13,19 @@
...
     "@near-js/crypto": "workspace:*",
     "@near-js/utils": "workspace:*",
-    "asn1-parser": "^1.1.8",
+    "asn1-parser": "1.1.8",
     "bn.js": "5.2.1",
     "borsh": "1.0.0",
-    "buffer": "^6.0.3",
-    "elliptic": "^6.5.4",
+    "buffer": "6.0.3",
+    "elliptic": "6.5.4",
```
Comment (paulmillr) [inline packages/biometric-ed25519/package.json:24]: why is elliptic used here in ed25519 context? elliptic's implementation of the curve is buggy and produces invalid outputs, which means in blockchain context "someone will lose money"

**Comment by frol** [comment]
From the library usage perspective, it is not great to pin all the dependencies down to a single version as that may block deduplicating the dependencies if two libraries of slightly different versions are used (1.0.1 and 1.0.2). I am wondering if there is some middle ground that we can find here.

Also, this PR only resolves the first part of #1200, so it will be too early to close that issue without reviewing and removing buggy dependencies.

**Comment by gtsonevv** [comment]
Hey guys, thank you for the replies. Based on them these are the next steps that we can take:
1. Our team will go through the dependency graph and resolve duplications.
2. We will try to reduce the bundle size. Looking at https://bundlephobia.com/package/near-api-js@2.1.4 we can see that 10% is bn.js which should be replaceable by native JS BigNumber. 7% is tweetnacl which is already replaced. We will try to find a comparable replacement for ajv because it’s biggest library by far(30%).

**Comment by paulmillr** [comment]
others:

- js-sha256 => noble-hashes (already used)
- elliptic => noble-curves (already used)
- borsch deps
    - remove bn.js (native BigInt)
    - remove text-encoding-utf-8 (native TextEncoder)
    - bs58 should be upgraded upstream to remove safe-buffer, or replaced with scure-base
- re-evaluate need for node-fetch
- re-evaluate need for error-polyfill

**Review by frol — APPROVED** [review]
As suggested offline, let's merge this PR with the pinned dependencies, and then work on a separate PR that will slim down the dependencies.

**Pull request merged**